os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -77,6 +77,51 @@
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+static void allocateregions(FAR struct elf_loadinfo_s *loadinfo)
+{
+	size_t sizes[3] = {loadinfo->textsize, loadinfo->rosize, loadinfo->binp->ramsize};
+	uintptr_t *allocs[3] = {&loadinfo->textalloc, &loadinfo->roalloc, &loadinfo->dataalloc};
+	uint32_t totalsize = sizes[0] + sizes[1] + sizes[2];
+	size_t tmpsz;
+	uintptr_t *tmpalloc;
+
+
+	if (sizes[0] < sizes[1]) {
```

Comment:
How about using bubble sort? We only have 3 members, so using bubble sort looks fine and simple.
```
for (i = 0; i < N; i++) {
    for (j = 0; j < N-(i+1); j++) {
        if (sizes[j] > sizes[j+1]) {
            temp = sizes[j+1];
            sizes[j+1] = sizes[j];
            sizes[j] = temp;

            temp = allocs[j+1];
            allocs[j+1] = allocs[j];
            allocs[j] = temp;
        }
    }
}
```

Reply:
Hi,
The current logic for comparison is the same as bubble sort but without the extra loop variables.

Reply:
Done.